tools/deploy_translatable_strings.sh

```diff
@@ -14,74 +14,94 @@
 
 # Script for pushing the translatable messages to poBranch.
 
+# DO NOT `set -x`.  We have to pass secrets to `openssl` on the command line,
+# and we don't want them appearing in the log.  This script instead manually
+# 'echo's its status at various points.
+set -eu -o pipefail
+
+if [[ "$#" -ne 1 ]]; then
+    echo "Usage: deploy_translatable_string.sh /path/to/translations/artifact" >&2
+    exit 1
+fi
 
 # Variables used by this script.
 # From github actions the docs/locale/en directory from the sphinx build
 # gets downloaded from the github actions artifacts as deploy directory
-TARGET_REPOSITORY="git@github.com:qiskit-community/qiskit-translations.git"
-SOURCE_DIR=`pwd`
-SOURCE_LANG='en'
-
-SOURCE_REPOSITORY="git@github.com:Qiskit/qiskit.git"
-TARGET_BRANCH_PO="main"
-DOC_DIR_PO="deploy"
-TARGET_DOCS_DIR_PO="docs/locale"
-
-echo "show current dir: "
-pwd
-
-echo "Setup ssh keys"
-pwd
-set -e
-# Add poBranch push key to ssh-agent
-openssl enc -aes-256-cbc -d -in ../tools/github_poBranch_update_key.enc -out github_poBranch_deploy_key -K $encrypted_deploy_po_branch_key -iv $encrypted_deploy_po_branch_iv
-chmod 600 github_poBranch_deploy_key
-eval $(ssh-agent -s)
-ssh-add github_poBranch_deploy_key
-
-# Clone to the working repository for .po and pot files
-popd
-pwd
-echo "git clone for working repo"
-git clone --depth 1 $TARGET_REPOSITORY temp --single-branch --branch $TARGET_BRANCH_PO
-pushd temp
 
+TARGET_REPO="git@github.com:qiskit-community/qiskit-translations.git"
+TARGET_REPO_BRANCH="main"
+
+SOURCE_TOOLS_DIR="$(dirname "$(realpath "$0")")"
+# Absolute paths to the git repository roots for the source repository (which
+# this file lives in) and where we're going to clone the target repository.
+SOURCE_REPO_ROOT="$(dirname "$SOURCE_TOOLS_DIR")"
+TARGET_REPO_ROOT="${SOURCE_REPO_ROOT}/_qiskit_translations"
+
+SOURCE_LANG="en"
+# Absolute paths to the source and target directories for the translations
+# files.  CI should feed the source in for us - it depends on the particulars of
+# how it was built in a previous job.  The target is under our control.
+SOURCE_PO_DIR="$1"
+TARGET_PO_DIR="${TARGET_REPO_ROOT}/docs/locale/${SOURCE_LANG}"
+
+# Add the SSH key needed to verify ourselves when pushing to the target remote.
+echo "+ setup ssh keys"
+eval "$(ssh-agent -s)"
+openssl enc -aes-256-cbc -d \
+    -in "${SOURCE_REPO_ROOT}/tools/github_poBranch_update_key.enc" \
+    -K "$encrypted_deploy_po_branch_key" \
+    -iv "$encrypted_deploy_po_branch_iv" \
+    | ssh-add -
+
+# Clone the target repository so we can build our commit in it.
+echo "+ 'git clone' translations target repository"
```

Comment:
Hah I like your solution for `set +x`, the `+` prefix to match that is a nice touch too

Reply:
Yeah, exactly what I had in mind hahaha.